dashboard/app/controllers/projects_controller.rb

```diff
@@ -509,6 +509,63 @@ def remix
     redirect_to action: 'edit', channel_id: new_channel_id
   end
 
+  # GET /projects/:project_type/:channel_id/submission_status
+  def submission_status
+    project_type = params[:project_type]
+    channel_id = params[:channel_id]
+    _, project_id = storage_decrypt_channel_id(params[:channel_id])
+    project = Project.find_by(id: project_id)
+    status = get_status(channel_id, project_type, project)
+    render(status: :ok, json: {status: status})
+  end
+
+  def get_status(channel_id, project_type, project)
```

Comment:
Could this logic be moved inside the `Project` model? It feels like something we could read like `project.submission_status`. Also there we've included `SharedConstants` and I think could drop the prefix to make this a little easier to read?

Reply:
+1, would be great to include some comments for context also since this seems to be the source of truth for how we compute project submission status

Reply:
Just wanted to jot down a couple thoughts I had on this that I think might clean up things a little bit (could totally be off base though), LMK if it'd be helpful to chat through this!

```
# A constant somewhere that maps submission statuses to error messages,
# either in shared constants or in this controller
error_map  = {
  PROJECT_SUBMISSION_STATUS[:ALREADY_SUBMITTED] => "Once submitted, a project cannot be submitted again.",
  ...
}
```

```
# Controller logic that checks if you can submit a project and returns appropriate error message
try
  authorize! :submit, project
catch CanCan::AccessDenied
  return render status: :forbidden, json: {error: error_map[project.submission_status]}
```

```
# Addition to ability.rb to specify which projects can submit
can :submit, Project do |project|
  project.submission_status != PROJECT_SUBMISSION_STATUS[:CAN_SUBMIT]
end
```

Reply:
Oops, had a boolean backwards. Fixed :)

Reply:
Awesome - thanks! updated.